src/Fixer/ReturnNotation/ReturnAssignmentFixer.php

```diff
@@ -354,4 +373,98 @@ private function clearIfSave(Tokens $tokens, int $index): void
 
         $tokens->clearTokenAndMergeSurroundingWhitespace($index);
     }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of the first token of a definition (lambda, anonymous class or match) or `null` if not an anonymous
+     */
+    private function isCloseBracePartOfDefinition(Tokens $tokens, int $index): ?int
+    {
+        $index = $tokens->findBlockStart(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+        $candidateIndex = $this->isOpenBraceOfLambda($tokens, $index);
+
+        if (null !== $candidateIndex) {
+            return $candidateIndex;
+        }
+
+        $candidateIndex = $this->isOpenBraceOfAnonymousClass($tokens, $index);
+
+        return $candidateIndex ?? $this->isOpenBraceOfMatch($tokens, $index);
+    }
+
+    /**
+     * @param int $index open brace index
+     *
+     * @return null|int index of T_NEW of anonymous class or `null` if not an anonymous
+     */
+    private function isOpenBraceOfAnonymousClass(Tokens $tokens, int $index): ?int
+    {
+        $index = $tokens->getPrevMeaningfulToken($index);
+
+        if ($tokens[$index]->equals(')')) {
```

Comment:
There might be other tokens after the `)`, e.g.:
```php
new class() implements Foo {}
```

Reply:
thanks, support added :+1: